surfer/viz.py

```diff
@@ -1064,19 +1064,21 @@ def add_data(self, array, min=None, max=None, thresh=None,
             if min is None:
                 min = array.min() if array.size > 0 else 0
             if max is None:
-                max = array.max() if array.size > 0 else 0
+                max = array.max() if array.size > 0 else 1
         else:
             if min is None:
                 min = 0
             if max is None:
-                max = np.abs(center - array).max() if array.size > 0 else 0
+                max = np.abs(center - array).max() if array.size > 0 else 1
         if mid is None:
             mid = (min + max) / 2.
+        _check_limits(min, mid, max, extra='')
 
         # Create smoothing matrix if necessary
         if len(array) < self.geo[hemi].x.shape[0]:
             if vertices is None:
-                raise ValueError("len(data) < nvtx: need vertices")
+                raise ValueError("len(data) < nvtx (%s < %s): need vertices"
```

Comment:
I know you haven't introduced it here, but I actually have no idea what "need vertices" means in this error message.